src/backend/distributed/transaction/transaction_management.c

```diff
@@ -302,6 +303,7 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			activeSetStmts = NULL;
 			CoordinatedTransactionUses2PC = false;
 			ExecutorLevel = 0;
+			PlannerLevel = 0;
```

Comment:
Would it be a bug if this is not reset to `0`? Wonder if an `Assert` is in place.